parser.js

```diff
@@ -176,7 +179,28 @@ function generateAST(code, options, additionalParsingContext) {
     const ast = program.getSourceFile(FILENAME);
 
     extra.code = code;
-    return convert(ast, extra);
+    return {
+        ast: convert(ast, extra),
+        program
+    };
+}
+
+/**
+ * Generate the `parserServices` object for a parse result
+ * @param   {Result} result The return value of `generateAST`
+ * @returns {Object}        The `parserServices` object
+ */
+function getServices(result) {
+    const typeChecker = result.program.getTypeChecker();
+    return {
+        program: result.program,
+        getTSNode: node => nodeUtils.map.get(node),
+        getESNode: tsNode => nodeUtils.reverseMap.get(tsNode),
+        typeChecker,
```

Comment:
As per my comment above about minimal API surface - why do we need a reference to the typeChecker, if we are already exposing the created Program?